transformer_engine/pytorch/fp8.py

```diff
@@ -64,13 +64,22 @@ def check_fp8_block_scaling_support() -> Tuple[bool, str]:
     return False, "FP8 block scaled GEMM requires Hopper and CUDA >= 12.9."
 
 
+def check_recipe_support(recipe: Recipe) -> None:
+    """Check if the given recipe is supported."""
+    recipe_supported = False
+    unsupported_reason = f"Unknown recipe: {recipe}."
+    if isinstance(recipe, (DelayedScaling, Float8CurrentScaling)):
+        recipe_supported, unsupported_reason = check_fp8_support()
+    elif isinstance(recipe, Float8BlockScaling):
+        recipe_supported, unsupported_reason = check_fp8_block_scaling_support()
+    elif isinstance(recipe, MXFP8BlockScaling):
+        recipe_supported, unsupported_reason = check_mxfp8_support()
+    assert recipe_supported, unsupported_reason
+
+
 def get_default_fp8_recipe() -> Recipe:
     """FP8 recipe with default args."""
     if check_mxfp8_support()[0]:
-        # This is a temporary restriction until MXFP8 is supported for all
-        # gemm layouts.
-        if get_device_compute_capability() >= (12, 0):
-            return Float8BlockScaling()
```

Comment:
I removed it because
1. `Float8BlockScaling` is not supported on SM120. It's only supported on Hopper.
2. This code is never accessed because `check_mxfp8_support()[0]` is `False` for SM120.

Reply:
Let's use `Float8CurrentScaling` as the default for SM 12.0 then. We had some discussion debating between FP8 CS and DSv3: https://github.com/NVIDIA/TransformerEngine/pull/1939#discussion_r2195912775

Pinging @ptrendx @ksivaman @sudhakarsingh27

Reply:
Added a condition for it.